# version: "3.8"

services:
  app:
    image: pihole/pihole:2024.03.2
    container_name: pihole
    hostname: pihole
    environment:
      - DHCP_ACTIVE=false
      - FTLCONF_IGNORE_LOCALHOST=false
      - FTLCONF_PIHOLE_PTR=HOSTNAMEFQDN
      - FTLCONF_REPLY_WHEN_BUSY=DROP
      - IPv6=false
      - PIHOLE_DNS_=8.8.8.8;1.1.1.1 # Trailing underscore [in the name] is intentional
      - QUERY_LOGGING=false
      - TZ=Europe/Bratislava
      - WEBPASSWORD_FILE=/.homelab/webpassword.txt
      - WEBTHEME=default-darker
    volumes:
      - ./data/etc-pihole:/etc/pihole:rw
      - ./data/etc-dnsmasq.d:/etc/dnsmasq.d:rw
      - ./private/webpassword.txt:/.homelab/webpassword.txt:ro
    healthcheck:
      test: curl --fail http://localhost/admin || exit 1
      start_period: 20s
      interval: 10s
      timeout: 2s
      retries: 1

  app-backup:
    image: pihole-backuper:homelab
    build:
      context: backuper/
    container_name: pihole-backup
    hostname: pihole-backup
    depends_on:
      app:
        condition: service_healthy
    env_file:
      - ./private/app-backup.env
    environment:
      - BROWSER_PATH=/usr/bin/chromium
      - URL=http://pihole
    volumes:
      - ./data/app-backup:/backup:rw
      - ./log/app-backup:/log:rw
